apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -336,6 +337,13 @@ type EnvoyListenerConfig struct {
 	// +optional
 	DisableMergeSlashes *bool `json:"disableMergeSlashes,omitempty"`
 
+	// DisableServerHeaderTransformation signifies Envoy will not modify the Server header.
+	// It has 3 possible values: OVERWRITE, PASS_THROUGH, APPEND_IF_ABSENT
+	//
+	// Contour's default is OVERWRITE.
+	// +optional
+	DisableServerHeaderTransformation *http.HttpConnectionManager_ServerHeaderTransformation `json:"disableServerHeaderTransformation,omitempty"`
```

Comment:
This field should be structured like [DNSLookupFamily](https://github.com/projectcontour/contour/blob/5e3b212a990ee7d52a744291e19ad4cd97af6e43/apis/projectcontour/v1alpha1/contourconfig.go#L528-L565), i.e. with a Contour-defined "enum" type. Please format the godoc similarly as well, and include an explanation on the behavior of each option.